hashintel/hash
# Add top context bar to pages

## 🌟 What is the purpose of this PR?



This PR adds a top context bar to pages which includes breadcrumbs and other action buttons (share, version and comments). It also updates the sidebar layout to make it easy to add fixed elements (like page notification banner and top context bar) at component level instead of layout level.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1202621375435414/1202700524579028) _(internal)_


## 🔍 What does this change?




- Adds a context bar to the top of each page which contains breadcrumbs
- Updates LayoutWithSidebar structure. It receives a `fullWidth` prop which determines if padding/max-width of the page should be handled by the layout or by the component. This makes it possible to have fixed elements like `TopContextBar` or `PageNotificationBanner` in the component. Previously we had to pass fixed elements as props to `getLayoutWithSidebar`


## ⚠️ Known issues


- Share, comments and version buttons don't perform any action at the moment.


## ❓ How to test this?



1.  Visit any page, type or entity
1.  Confirm the context bar is present at the top and the breadcrumbs show the right info

## 📹 Demo



https://user-images.githubusercontent.com/16928688/184594255-c20017e3-63bd-4cd6-97fa-3b2c071c545b.mov




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202700524579028
